src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -109,5 +110,35 @@ private static Interop.libproc.rusage_info_v3 GetCurrentProcessRUsage()
         {
             return Interop.libproc.proc_pid_rusage(Environment.ProcessId);
         }
+
+        private static TimeSpan MapTime(ulong sysTime)
+        {
+            uint denom = s_timeBase_denom;
+            if (denom == default)
+            {
+                Interop.libSystem.mach_timebase_info_data_t timeBase = GetTimeBase();
+                s_timeBase_denom = denom = timeBase.denom;
+                s_timeBase_numer = timeBase.numer;
```

Comment:
```suggestion
                s_timeBase_numer = timeBase.numer;
                s_timeBase_denom = denom = timeBase.denom;
```
This has a race condition. Consider what happens if the thread is rescheduled right after `s_timeBase_denom` is assigned and before `s_timeBase_numer` is assigned; and some other thread executes this method. The `denom == default` condition is going to be false on the other thread and it will end up uninitialized `s_timeBase_numer`.

Reply:
Yep, you are right. No idea how we missed that during the initial review :(

What is the process to apply the fix for this race to both `master` and now to .NET 8 as well?

Reply:
Submit a PR with the fix to against the main branch. We can then cherry pick it into this one.

Reply:
Cherry-picked the fix from main